llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -602,19 +609,30 @@ def __init__(self, config: MPTConfig):
                     )
             self.logit_scale = logit_scale
 
-    def get_input_embeddings(self) -> nn.Embedding:
-        return self.transformer.wte
+    def get_input_embeddings(self) -> Union[SharedEmbedding, nn.Embedding]:
+        return self.transformer.get_input_embeddings()
 
     def set_input_embeddings(
             self, value: Union[SharedEmbedding, nn.Embedding]) -> None:
-        self.transformer.wte = value
+        self.transformer.set_input_embeddings(value)
 
-    def get_output_embeddings(self) -> nn.Embedding:
-        return self.transformer.wte
+    def get_output_embeddings(
+            self) -> Union[SharedEmbedding, nn.Embedding, nn.Linear]:
+        return self.lm_head or self.transformer.get_input_embeddings()
```

Comment:
```suggestion
        if self.lm_head is not None:
            return self.lm_head
        return self.transformer.get_input_embeddings()
```
would prefer this to be more explicit